### PR TITLE
Add new PV IsInjecting-Mon 

### PR DESCRIPTION
**Pull request by anacso17**
This PR adds new IsInjecting-Mon PV to InjCtrl IOC: a flag that indicates whether an injection event is occurring. Also, this PR adds related IsInjDelay-SP/RB and IsInjDuration-SP/RB PVs to configure flag delay and duration.

### CODE REVIEW

**Review by mmdonatti — APPROVED** [review]

File: siriuspy/siriuspy/injctrl/main.py [hunk 1]
```diff
@@ -1050,6 +1096,29 @@ def _callback_update_pu_refvolt(self, pvname, value, **kws):
         index = self._pu_names.index(devname)
         self._pu_refvolt[index] = value
 
+    def _callback_is_injecting(self, value, **kws):
+        if value is None:
+            return
+
+        # check if any InjSI PU is pulsing
```
Comment (fernandohds564) [inline siriuspy/siriuspy/injctrl/main.py:1103]: I think it is better to delegate all this to a new thread.

File: siriuspy/siriuspy/injctrl/main.py [hunk 2]
```diff
@@ -1050,6 +1096,29 @@ def _callback_update_pu_refvolt(self, pvname, value, **kws):
...
+        is_injecting = False
+        for pvs in self._pvs_injsys.values():
+            if not all([pvo.connected for pvo in pvs]):
+                break
```
Comment (fernandohds564) [inline siriuspy/siriuspy/injctrl/main.py:1107]: I think if they are not connected we must assume the worst scenario.

File: siriuspy/siriuspy/injctrl/main.py [hunk 3]
```diff
@@ -1050,6 +1096,29 @@ def _callback_update_pu_refvolt(self, pvname, value, **kws):
...
+            pvs_vals = [pvo.value for pvo in pvs]
+            if all(pvs_vals):
```
Comment (fernandohds564) [inline siriuspy/siriuspy/injctrl/main.py:1109]: I think you don't need this variable `pvs_vals`. You could simply do:
```python3
if all(pvo.value for pvo in pvs)
```

File: siriuspy/siriuspy/injctrl/main.py [hunk 4]
```diff
@@ -1050,6 +1096,29 @@ def _callback_update_pu_refvolt(self, pvname, value, **kws):
...
         self._pu_refvolt[index] = value
 
+    def _callback_is_injecting(self, value, **kws):
+        if value is None:
+            return
+
+        # check if any InjSI PU is pulsing
+        is_injecting = False
+        for pvs in self._pvs_injsys.values():
+            if not all([pvo.connected for pvo in pvs]):
```
Comment (fernandohds564) [inline siriuspy/siriuspy/injctrl/main.py:1106]: I think you don't need to encapsulate the generator in a list comprehension. You could do:
```python3
if not all(pvo.connected for pvo in pvs)
```
without the `[...]`.

File: siriuspy/siriuspy/injctrl/main.py [hunk 5]
```diff
@@ -1050,6 +1096,29 @@ def _callback_update_pu_refvolt(self, pvname, value, **kws):
...
+                break
+            pvs_vals = [pvo.value for pvo in pvs]
+            if all(pvs_vals):
+                is_injecting = True
+                break
+
+        # if True, raise IsInjecting flag after {delay}ms for {duration}ms
+        if is_injecting:
```
Comment (fernandohds564) [inline siriuspy/siriuspy/injctrl/main.py:1114]: maybe the construction
```python3
if not is_injecting:
    return

# rest of the code
```
is cleaner, isn't it?

**Review by fernandohds564 — CHANGES_REQUESTED** [review]

**Review by xresende — APPROVED** [review]

File: siriuspy/siriuspy/injctrl/main.py (hunk 2 above)
Comment (anacso17) [reply siriuspy/siriuspy/injctrl/main.py:1111]: done!

File: siriuspy/siriuspy/injctrl/main.py (hunk 3 above)
Comment (anacso17) [reply siriuspy/siriuspy/injctrl/main.py]: done!

File: siriuspy/siriuspy/injctrl/main.py (hunk 5 above)
Comment (anacso17) [reply siriuspy/siriuspy/injctrl/main.py]: done!

**Review by fernandohds564 — APPROVED** [review]

**Pull request merged**